### ISSUE TITLE
Potential Bug for Function 'LinkNeighborLoader' 

### ISSUE THREAD

**Issue by hktikhin** [pyg-team/pytorch_geometric#5918]
### 🐛 Describe the bug

Background:
I want to build an anime recommendation system using gnn technique.
I split my pyg graph data into train/ dev/ test set.
And then do negative sampling and mini-batch sampling for the train data.
Here's my code.
```
# Define seed edges:
edge_label_index = train_data["user", "like", "anime"].edge_label_index
edge_label = train_data["user", "like", "anime"].edge_label
train_loader = LinkNeighborLoader(
    data=train_data,
    num_neighbors=[20, 10], 
    neg_sampling_ratio=2,
    edge_label_index=(("user", "like", "anime"), edge_label_index),
    edge_label=edge_label,
    batch_size=batch_size,
    shuffle=True,
)
```

Total # of user node is `306492` and Total # of anime node is `16343`
I use this code to get the batch and the edge label index used for link prediction.

```
sampled_data = next(iter(train_loader))
sampled_data['like'].edge_label_index
```

```
output: 
tensor([[ 353, 1378,  988,  ..., 4110, 4294, 2531],
        [   9, 4463, 2577,  ...,  978,  527, 2955]])
# issue: edge_label_index in batch loader doesn't give me back the index corresponding to the graph
# it is actually being reindexed 
```
The problem is that when I call  `sampled_data['like'].edge_label_index`, I expect to get the souce node index ranging from 0 - 306492 and the target node index ranging from 0 - 16343.
But it is actually being reindexed.
And when I use these edge index to make prediction, I cann't retrieve back the corresponding user and anime feature.

### Environment

* PyG version:
* PyTorch version:
* OS:
* Python version:
* CUDA/cuDNN version:
* How you installed PyTorch and PyG (`conda`, `pip`, source):
* Any other relevant information (*e.g.*, version of `torch-scatter`):

**Comment by rusty1s** [comment]
If you want to obtain your global node ids, you can do something like this
```python
train_data["user"].node_id = torch.arange(train_data["user].num_nodes)
train_data["anime"].node_id = torch.arange(train_data["anime"].num_nodes)

loader = ...

global_row = sampled_data["user"].node_id[sampled_data["user", "anime"].edge_label_index[0]]
global_col = sampled_data["anime"].node_id[sampled_data["user", "anime"].edge_label_index[1]]
global_edge_label_index = torch.stack([global_row, global_col], dim=0)
